pkg/webhook/stats_reporter.go

```diff
@@ -17,6 +17,8 @@ const (
 	validationRequestCountMetricName    = "validation_request_count"
 	validationRequestDurationMetricName = "validation_request_duration_seconds"
 
+	validationRequestPending = "validation_request_pending"
```

Comment:
can we add suffix `_count` to be consistent with other metrics? 
https://prometheus.io/docs/practices/naming/#metric-names

Reply:
This metric is a gauge - it can go up and down, as opposed to a counter. I believe the other metrics you're referring to might be counters? My understanding of Prom metric naming best practices is that `_total` is the recommended suffix for counters, while `_count` is used for the counter portion of histograms. I'm not aware of a standard suffix for gauges, and if you go by [this writeup](https://www.robustperception.io/on-the-naming-of-things):

> A Gauge ending with _count or _total is quite confusing, as it implies it's a Counter rather than a Gauge.

Thoughts?

On a related note - the main things that I'm uncomfortable with now are the plurality (`validation_requests_pending`?) and whether or not we move to using labels to differentiate endpoints (`requests_pending{endpoint="/v1/admit"}`).

Reply:
+1 to not using `_count` on a gauge.

IIRC, we didn't come to a final conclusion WRT the metrics naming. I think you were willing to come up with a doc focusing on metric naming schemes and backwards compatibility concerns?

One thing I have against the `endpoint` label specifically is that it's not semantic (do users care if we're using the v2 API?), though splitting by request type does make sense.

Reply:
ah you are right, we are using `_count` for counters but other gauges don't have that suffix.

Reply:
^- fwiw it looks like the convention of `_count` being reserved for histograms is baked into the `promlint` utility:

https://github.com/prometheus/client_golang/blob/master/prometheus/testutil/promlint/promlint.go#L203-L205